python/tests/transformers/tf_tensor_test.py

```diff
@@ -0,0 +1,236 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import shutil
+import tempfile
+
+from keras.layers import Conv1D, Dense, Flatten, MaxPool1D
+import numpy as np
+import tensorflow as tf
+import tensorframes as tfs
+
+from pyspark.sql.types import Row
+
+from sparkdl.graph.builder import IsolatedSession
+import sparkdl.graph.utils as tfx
+from sparkdl.transformers.tf_tensor import TFModelTransformer
+
+from ..tests import SparkDLTestCase
+
+def grab_df_arr(df, output_col):
+    """ Stack the numpy array from a DataFrame column """
+    return np.array([row.asDict()[output_col]
+                     for row in df.select(output_col).toLocalIterator()])
+
+class TFModelTransformerTest(SparkDLTestCase):
+
+    def _get_rand_vec_df(self, num_rows, vec_size):
+        return self.session.createDataFrame(
+            Row(idx=idx, vec=np.random.randn(vec_size).tolist())
+            for idx in range(num_rows))
+
+    def test_checkpoint_reload(self):
+        vec_size = 17
+        num_vecs = 31
+        df = self._get_rand_vec_df(num_vecs, vec_size)
+        analyzed_df = tfs.analyze(df)
+        input_col = 'vec'
+        output_col = 'outputCol'
+
+        # Build the TensorFlow graph
+        model_temp_dir = tempfile.mkdtemp()        
+        ckpt_dir = os.path.join(model_temp_dir, 'model_ckpt')
+        with tf.Session() as sess:
+            x = tf.placeholder(tf.float64, shape=[None, vec_size], name='tnsrIn')
+            w = tf.Variable(tf.random_normal([vec_size], dtype=tf.float64),
+                            dtype=tf.float64, name='varW')
+            z = tf.reduce_mean(x * w, axis=1, name='tnsrOut')            
+            sess.run(w.initializer)
+            saver = tf.train.Saver(var_list=[w])
+            saved_path = saver.save(sess, ckpt_dir, global_step=2702)
```

Comment:
Save the model and record the current `global_step`. Then we'd expect files like these.
```
model.ckpt-2702.index
model.ckpt-2702.meta
model.ckpt-2702.data-00000-of-00001
```